File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        l=[]
        for i in range(len(nums)):
            pre = 1
            suf = 1
            for j in range(i):
                pre = pre * nums[j]
            for j in range(i+1,len(nums)):
                suf = suf * nums[j]
            l.append(pre * suf)
        return l
        

File: Data Structures & Algorithms/first-missing-positive/submission-1.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int i = 0, n = nums.size();
        while (i < n){
            int idx = nums[i] - 1;
            if (1 <= nums[i] && nums[i] <= n && nums[i] != nums[idx]){
                swap(nums[idx], nums[i]);
            }
            else{
                i++;
            }
        }
        for (int j = 0 ; j < n ; j++){
            if (nums[j] != j + 1){
                return j + 1;
            }
        }
        return n + 1;
    }
};